hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/utils/MiscCryptoUtils.java

```diff
@@ -45,11 +55,20 @@ public static byte[] keccak256DigestOf(final byte[] msg) {
      * @return the raw bytes of the public key coordinates
      */
     public static byte[] decompressSecp256k1(final byte[] compressedKey) {
-        final var pk = curveSecp256k1.decodePoint(compressedKey);
-
+        // convert public key to native format
+        final LibSecp256k1.secp256k1_pubkey pubkey = PUB_KEY_CACHE.get();
+        final int keyParseResult = LibSecp256k1.secp256k1_ec_pubkey_parse(
+                LibSecp256k1.CONTEXT, pubkey, compressedKey, compressedKey.length);
+        if (keyParseResult != 1) throw new RuntimeException("Failed to parse public key");
+        final ByteBuffer outputBuffer = UNCOMPRESSED_PUBLIC_KEY_BYTEBUFFER_CACHE.get();
```

Comment:
byte array from `UNCOMPRESSED_PUBLIC_KEY_INPUT_CACHE` could be reused for `ByteBuffer outputBuffer`.